File: 001_two_sum/two_sum.py
class Solution:
    def twoSum(self, nums, target):
        h = {}
        for i, n in enumerate(nums):
            other = target - n
            if other not in h:
                h[n] = i
            else:
                return [h[other], i]

print(Solution().twoSum([-1, -2, -3, -4, -5], -8))

File: 0001_two_sum/two_sum.cc
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> res;
        unordered_map<int, int> ht; 
        for (int i = 0; i < nums.size(); i++) {
            int other = target - nums[i];
            if (ht.find(other) != ht.end()) {
                /* Only one solution for purpose of this problem */
                res.append(ht[other]);
                res.append(i);
                return res;
            }
            ht[nums[i]] = i;
        }
        return res;
    }
};
